department-of-veterans-affairs/vets-api
# API-29158 Backport download endpoint to Decision Reviews v2

## Summary

A `download` endpoint was previously added only to the segmented APIs for 200995s, 200996s, and 10182s. We've decided it should also be available in Decision Reviews, so this PR backports those endpoints to the following decision reviews routes:

* `/services/appeals/v2/decision_reviews/notice_of_disagreements/:id/download`
* `/services/appeals/v2/decision_reviews/higher_level_reviews/:id/download`
* `/services/appeals/v2/decision_reviews/supplemental_claims/:id/download`

These endpoints all expect an `X-VA-ICN` header to check against the saved data on the appeal. I chose to ingest this a header instead of a param like in the segmented APIs because there's already a method available in the decision reviews controllers to validate it. 

This PR does **NOT** include docs yet. I have some work in progress for docs and will make a follow-up PR for them, plus a PR for release notes.

## Related issue(s)
- [API-29158](https://jira.devops.va.gov/browse/API-29158)

## Testing done

- Manually tested the endpoints with a local server
  - If you want to test this locally, **note that** you can only download a PDF for an appeal if:
    1. it's in a 'submitted' state (or later)
    2. it has a 'pdf_version' set ('v3' is the default - this value is set on submission in real life, but when testing locally it must be set manually)

## What areas of the site does it impact?
* Adds a new `download` endpoint to higher level reviews, notice of disagreements, and supplemental claims areas of the Decision Reviews v2 API

## Acceptance criteria
